code/addons/storysource/src/StoryPanel.tsx

```diff
@@ -112,9 +112,11 @@ export const StoryPanel: React.FC<StoryPanelProps> = ({ api }) => {
   };
 
   const createParts = ({ rows, stylesheet, useInlineStyles }: SyntaxHighlighterRendererProps) => {
-    const parts = [];
+    const parts: React.ReactNode[][] = [];
     let lastRow = 0;
```

Comment:
Are you sure that createElement returns an array ReactNode's? Because I think it just a ReactNode, and I think this type should be therefore `ReactNode[]`

Reply:
@kasperpeulen Can we do a double check for this?

This is the logic for `createPart`:
```js
  const createPart = ({ rows, stylesheet, useInlineStyles }: SyntaxHighlighterRendererProps) =>
    rows.map((node, i) =>
      createSyntaxHighlighterElement({
        node,
        stylesheet,
        useInlineStyles,
        key: `code-segment${i}`,
      })
    );
```

`createSyntaxHighlighterElement` is the `createElement` renamed, which returns ReactNode. Seems like `rows` is an array of something (it's literally defined as any at `SyntaxHighlighterRendererProps` at @storybook/components)

Therefore, I think `createPart` does return the array of ReactNode.

Let me know if I'm missing anything 🤔

Reply:
Ah you are right! I think I would then do:
```ts
parts.push(...start);
parts.push(storyPart);
```
To have it all in one array with unique react keys. 

Can you also type the return type of the `createPart` function to be `ReactNode[]` so it is more clear what is going on 
 for the next reader 😅

Reply:
Got it. Will do